proxy/src/lib.rs

```diff
@@ -229,7 +222,7 @@ where
             let fut = serve(
                 outbound_listener,
                 outgoing,
-                tcp_connect_timeout,
+                config.private_connect_timeout,
```

Comment:
This should be `public_connect_timeout`, right? The code previously passed `public_connect_timeout` when constructing the `Bind`. Also, `private_connect_timeout` is the one that's way too low, right?

Reply:
Also, assuming this PR is correct, we should probably remove the TCP connect timeout too? That is, I don't know why we'd have a timeout for non-HTTP traffic but not have one for HTTP traffic.

Reply:
> Also, assuming this PR is correct, we should probably remove the TCP connect timeout too?

In a follow-up PR, assuming this one is approved, I mean.